### DISCUSSION TITLE
html pop up accessory view - support for CSS

### DISCUSSION THREAD

**Discussion by mm2270** [IBM/mac-ibm-notifications#64] [Q&A] [answered]
Hi, I've been trying to use the html accessory view type for something, and some of the styling I'd like to do is easier or more possible to do using CSS. But in my tests, it doesn't seem like IBM Notifier supports this. It looks like it may only accept normal html. Is this the case, or am I missing something? I have html + css set up that shows up correctly in a browser, but using those same settings passed to IBM Notifier doesn't respect any of the CSS. For example, resizing an image with CSS, the image comes in full size instead of scaled to the size I specified.

Just wanted to make sure I'm not overlooking something here. If the product doesn't support that, is there any way something like that could be added in a future version?

**Comment by SMartorelli** [comment, marked as answer]
Hi @mm2270, IBM Notifier accept only simple HTML. Can you provide some example UIs of what are you trying to show to the user? If so I can try to suggest you a possible solution or to open a feature request if the result could be achieved with some code adjustment.

**Reply by mm2270** [reply]
Hi @SMartorelli !

So yes, I can provide some examples, but generally speaking one of the items I notice is an issue is scaling an image with html.

In my example, I want to display a banner type image across the top of the pop up window and then html styled text below it. I can style the text no issue, but when using any html formatting for the image on the `<img src=` line such as `width="100" height="50"` as an example, IBM Notifier ignores this and displays the image at full size. It comes in large and I only see a small amount of the image. The rest gets cut off.
I've also tried using `width=20%"` as an example, which is also valid, but doesn't affect the image. The only way I've been able to get an image to display at a size relative to the pop up window is to pre-scale the image down to the exact size I need it.

What's odd is that IBM Notifier definitely seems to know how to scale an image to fit the popup since it automatically does this when using the `image` accessory view type. I'm not sure why it doesn't respect the width and height values that ride along with the image when using the html accessory type.
It was at that point I attempted to use CSS styling as a workaround and saw that it doesn't use any CSS. I'm not bothered by this since I imagine supporting CSS is much harder. But it would be nice if it supported some more of the known html attributes.

This honestly is a pretty minor issue, all things considered. It's awesome that I can even use html in the pop up to begin with, but it would be cool if I didn't have to craft a specially sized image to use in an html pop up view.

**Reply by SMartorelli** [reply]
Ok I got what you're trying to do! sorry but IBM Notifier doesn't really support images embedded in the HTML payload. The `html` and `htmlwhitebox` accessory views serves as formatting tool for text only payloads, I suggest you to use it only for text representation and use the `image` accessory view to show images.

**Reply by mm2270** [reply]
Ok thanks @SMartorelli!
The thing is, IBM Notifier actually does support images embedded in the `html` and `htmlwhitebox` accessory views, even if that wasn't the original intention. I've been able to get it to work, but as mentioned, the image must be the correct size or it displays all weird.
I guess I'll explore some other options on this, or just keep using it as I mentioned with the properly scaled image.

**Reply by SMartorelli** [reply]
Have you tried to use two different accessory views for image and text? Now you can use a normal `-accessory_view_type/payload` and a `-secondary_accessory_view_type/payload`

**Reply by mm2270** [reply]
Yes, I have tried that. The problem is, I need to use the `-secondary_accessory_view_type` for the `timer` option in my particular example, not an image. So I can't use all 3 (image, html, timer), that's why I was trying to add an image to the html as the primary view and then use the timer as the secondary accessory view.

It's ok though. I'll manage to work it out. Thanks!

**Reply by SMartorelli** [reply]
Oh gotcha, last thing I can suggest to you is to use the subtitle for the text, it supports markdown format, it is not HTML but better than plain text haha

**Reply by mm2270** [reply]
Thank you. I actually didn't know the `subtitle` text supported Markdown. That's very good to know. I've been playing with it.

I may post a request. It looks like some lesser used Markdown constructs aren't properly supported, but most of the main ones seem to be.
